homeassistant/components/deconz/climate.py

```diff
@@ -190,6 +191,21 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
                 mode=HVAC_MODE_TO_DECONZ[hvac_mode],
             )
 
+    @property
+    def hvac_action(self) -> str | None:
+        """Return current hvac operation ie. heat, cool.
+
+        Need to be one of HVACAction.*.
+        """
+        if self._device.mode == ThermostatMode.OFF:
+            return HVACAction.OFF
+
+        if self._device.state_on or self._device.preset == ThermostatPreset.BOOST:
```

Comment:
Im not sure I follow the reasoning with preset boost would consider the device to be active, what's different from the other presets and why wouldn't state_on be true if its active?

Reply:
The preset "boost" is only active if it is currently boosting (heating / cooling at 100%)
There is no situation where the preset will be boost but the device is not heating / cooling at 100%

If the state_os is True, the device is currently doing whatever the mode is set to (e.g. heating / cooling).

Thats how Smanar explained it to me and I implemented a thermostate device.

Works perfectly fine with all existing test cases in test_climate.py

Reply:
Wouldn't stat_on be true when preset is boost?

Reply:
Yes, this would be the ideal case, but I think the device I have / not all devices do not change the state to "on" if it's boosting.
I will add the description and a test case later today.